source/appModules/soffice.py

```diff
@@ -101,6 +105,37 @@ def _getFormatFieldAndOffsets(self,offset,formatConfig,calculateOffsets=True):
 		if backgroundColor:
 			formatField['background-color']=colors.RGB.fromString(backgroundColor)
 
+		if offset == 0:
+			# Only include the list item prefix on the first line of the paragraph.
+			numbering = formatField.get("Numbering")
+			if numbering:
+				formatField["line-prefix"] = numbering.get("NumberingPrefix") or numbering.get("BulletChar")
+
+		return formatField
+
+	def _getFormatFieldAndOffsets(self, offset, formatConfig, calculateOffsets=True):  # noqa: C901

```

Comment:
Given C901 - could you ensure that we are reducing the function complexity in this change?
This could be done by factoring out the added code into a separate helper function

Reply:
I'm not sure how to address this, can can you please be more specific what you want me to move?

 The only thing added is this block:

```
		# LibreOffice >= 24.2 uses IAccessible2 text attributes, earlier versions use
		# custom attributes, with the attrtibutes string starting with "Version:1;"
		if attribsString and attribsString.startswith('Version:1;'):
			formatField = self._getFormatFieldFromLegacyAttributesString(
				attribsString,
				offset
			)
		else:
			formatField, (startOffset, endOffset) = super()._getFormatFieldAndOffsets(
				offset,
				formatConfig,
				calculateOffsets
			)
```
, for which I'm not sure whether it really makes things more readable to move that to a separate method.

In my eyes, the change already reduces complexity, since it already splits most of the existing method into a helper method (for the legacy handling). As mentioned in the description, for me, the easiest way to simplify that further would be by dropping the legacy handling once it's no more needed. Is there a guideline/policy on whether/how that's done, (e.g. can that be dropped half a year after an old version is end-of-life?), or does compatibility with other versions have to be maintained forever?

Reply:
Thanks for the explanation, @lukaszgo1 ! No, there's no systematic code exchange between LibreOffice and OpenOffice. My personal impression (also from a quick look at the git log)  is that OpenOffice is currently mostly in some kind of maintenance mode, so I currently wouldn't expect any larger changes regarding accessibility there.

Reply:
A C901 indicates that code is too complex - long functions are always a readability issue as well as a general complexity issue, as it leads to methods that perform multiple logical actions. 
I would encourage moving that new block into a helper function.
With a large amount of inline code, i.e. a C901, removing groupable chunks into helper functions 
The overall goal with a C901 is to reduce code complexity of the whole function.
This PR essentially introduces a new C901, while improving an existing one.

When adding noqa comments for C901, please see other examples e.g.
This indicates the procedure we follow to reduce C901 issues in the code base.

```
# C901 'setFocusObject' is too complex
# Note: when working on setFocusObject, look for opportunities to simplify
# and move logic out into smaller helper functions.
```

Reply:
Thanks, I've extracted another helper method now with the new block of code and everything that was above it and added another comment for the C901 for the method handling the legacy attributes.